### PR TITLE
Add bootstrapping

### LINKED ISSUE #203: Feat: add support for bootstrapping
Improve resilience by implementing the Bootstrap feature of Unleash. Feature write-up:
https://www.getunleash.io/blog/bootstrap-unleash-java-sdk-to-improve-the-resilience

Reference implementation: 
* Java SDK: https://github.com/Unleash/unleash-client-java/pull/131
* Node SDK: https://github.com/Unleash/unleash-client-node/pull/283/files

We need to bootstrap to be adapted to what makes sense in Pyhton. It should at least allow:
- manual passing of "json string of the full feature toggle configuration", 
- loading from file 
- loading from URL. 

If the user want to bootstrap from a URL we should probably try he actual Unleash API, and load the bootstrap URL as a fallback mechanism of the SDK fails to load it from the Unleash API. It will only load from bootstrap on the initial fetch.

### PR DESCRIPTION
**Pull request by ivanklee86**
# Description

- Add support for bootstraping UnleashClient from dictionary, file, or url.
 
Fixes #203 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

- [x] Unit tests
- [x] Spec Tests
- [x] Integration tests / Manual Tests

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: UnleashClient/cache.py [hunk 1]
```diff
@@ -27,9 +31,73 @@ def destroy(self):
 
 
 class FileCache(BaseCache):
+    """
+    The default cache for UnleashClient.  Uses `fcache <https://pypi.org/project/fcache/>`_ behind the scenes.
+
+    You can boostrap the FileCache with initial configuration to improve resiliancy on startup.  To do so:
```
Comment (sighphyre) [inline UnleashClient/cache.py:37]:
```suggestion
    You can boostrap the FileCache with initial configuration to improve resiliency on startup.  To do so:
```

**Review by sighphyre — COMMENTED** [review]
This looks pretty great.

One question though, if I'm reading the code right, if the user bootstraps the cache and then passes in `False` for `bootstrapped` the cache will get overwritten with the next fetch result without being used. Is there a valid use case for this? Otherwise it might be harder to misuse if 
`bootstrap_from_dics`/`bootstrap_from_file`/`bootstrap_from_url` set a bootstrapped flag internally and the Unleash client can read that. WDYT?

**Comment by ivanklee86** [comment]
@sighphyre I think I've manded to do it in a backwards compatible way. =) `bootstrapped` is now a BaseCache attribute and will be set by the FileCache bootstrapped* methods.

**Review by sighphyre — APPROVED** [review]
Nice!

**Comment by sighphyre** [comment]
@ivanklee86 Ah, awesome, you legend. This looks great, I'm happy if you're happy

**Pull request merged**